main.c

```diff
@@ -411,6 +370,39 @@ static asm_return_t lz_linux(struct tpm *tpm, struct lz_tag_boot_linux *lz_tag)
 	print("and again2\n");
 	hexdump(_p(0), 0x30);
 #endif
+}
+
+/*
+ * Function return ABI magic:
+ *
+ * By returning a simple object of two pointers, the SYSV ABI splits it across
+ * %rax and %rdx rather than spilling it to the stack.  This is far more
+ * convenient for our asm caller to deal with.
+ */
+typedef struct {
+	void *pm_kernel_entry; /* %eax */
+	void *zero_page;       /* %edx */
+} asm_return_t;
+
+static asm_return_t lz_linux(struct tpm *tpm, struct lz_tag_boot_linux *lz_tag)
+{
+	struct boot_params *bp;
+	struct kernel_info *ki;
+	struct mle_header *mle_header;
+	void *pm_kernel_entry;
+
+	/*
+	 * Now in 64b mode, paging is setup. This is the launching point. We can
+	 * now do what we want. First order of business is to setup IOMMU to cover
+	 * all memory. At the end, trampoline to the PM entry point which will
+	 * include the Secure Launch stub.
+	 */
```

Comment:
This comment is obsolete.  Amongst other things, we might be in 32bit mode.  Seeing as the hunk has moved by virtue of the introduction of iommu_setup(), I'd just drop it.